src/gui/entry/EntryModel.cpp

```diff
@@ -274,36 +274,43 @@ QVariant EntryModel::data(const QModelIndex& index, int role) const
             break;
         case Title:
             return Icons::entryIconPixmap(entry);
-        case Paperclip:
-            if (!entry->attachments()->isEmpty()) {
-                return icons()->icon("paperclip");
-            }
-            break;
-        case Totp:
-            if (entry->hasTotp()) {
-                return icons()->icon("totp");
-            }
-            break;
+
         case PasswordStrength:
             if (!entry->password().isEmpty() && !entry->excludeFromReports()) {
+                QString iconName = "lock-question";
                 StateColorPalette statePalette;
                 QColor color = statePalette.color(StateColorPalette::Error);
 
                 switch (entry->passwordHealth()->quality()) {
                 case PasswordHealth::Quality::Bad:
                 case PasswordHealth::Quality::Poor:
+                    iconName = "lock-open-alert";
                     color = statePalette.color(StateColorPalette::HealthCritical);
                     break;
                 case PasswordHealth::Quality::Weak:
+                    iconName = "lock-open";
                     color = statePalette.color(StateColorPalette::HealthBad);
                     break;
                 case PasswordHealth::Quality::Good:
                 case PasswordHealth::Quality::Excellent:
+                    iconName = "lock";
                     color = statePalette.color(StateColorPalette::HealthExcellent);
                     break;
                 }
 
-                return color;
+                if (color.isValid()) {
+                    return icons()->icon(iconName, true, color);
+                }
+            }
+            break;
+        case Paperclip:
```

Comment:
Are moved `Paperclip` and `Totp` **cases** from line 277 to 306 related to the password strength icon implementation? If not, I would keep them `as is`, otherwise it confuses a little.

Reply:
Changed it back!